apps/meteor/app/lib/server/lib/notifyUsersOnMessage.ts

```diff
@@ -41,16 +33,19 @@ export async function getMentions(message) {
 	const toAll = mentions.some(({ _id }) => _id === 'all');
 	const toHere = mentions.some(({ _id }) => _id === 'here');
 
-	const userMentions = mentions.filter((mention) => !mention.type || mention.type === 'user');
-	const otherMentions = mentions.filter((mention) => mention?.type !== 'user');
-
-	const filteredMentions = userMentions.filter(({ _id }) => _id !== senderId && !['all', 'here'].includes(_id)).map(({ _id }) => _id);
-
-	const mentionIds = await callbacks.run('beforeGetMentions', filteredMentions, {
-		userMentions,
-		otherMentions,
-		message,
-	});
+	let teamsMentions = [] as any[];
```

Comment:
instead of populating teamsMentions in the filter, can't you just populate it here?
```typescript
const teamsMentions = mentions.filter((mention) => mention.type === 'team');
const filteredMentions = mentions
		.filter((mention) => !mention.type || mention.type === 'user')
		.filter(({ _id }) => _id !== senderId && !['all', 'here'].includes(_id))
		.map(({ _id }) => _id);
```

Reply:
Got it. For the sake of code maintainability I changed that adding a specific filter to teamsMentions (https://github.com/RocketChat/Rocket.Chat/pull/31877/commits/2c18390eac5d41ddaa23faa61ff03bc1bc0f9126).